app/models/guest_device.rb

```diff
@@ -10,4 +11,7 @@ class GuestDevice < ApplicationRecord
 
   has_one :network, :foreign_key => "device_id", :dependent => :destroy, :inverse_of => :guest_device
   has_many :miq_scsi_targets, :dependent => :destroy
+
+  has_many :firmwares, :dependent => :destroy
+  has_many :child_devices, -> { where(:parent_device_id => ids) }, :foreign_key => "parent_device_id", :class_name => "GuestDevice", :dependent => :destroy
```

Comment:
K... Thanks for the update @skovic ... I'll remove my "request for changes" review and take another look.

We can try to figure out what's going on with this in a separate PR/thread.